operator/pkg/gateway-api/helpers/serviceimport.go

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package helpers
+
+import (
+	"context"
+	"fmt"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/types"
+	mcsapiv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+	mcsapicontrollers "sigs.k8s.io/mcs-api/pkg/controllers"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var hasServiceImportCRD bool = false
```

Comment:
On second thought it would be really nice if we could get rid of this global var.

When I think about it, it would be enough to do the following things:

1. Pass information (bool) about enabled ServiceImport functionality (retrieved via `HasServiceImportCRD` (consolidate with `CheckServiceImportCRD` - as we no longer need the global var) down into `registerReconcilers`
2. Pass the information about the enabled MCS `ServiceImport` functionality into the various reconcilers that need this information (`grpc`, `http`, `tls`) -> they can keep the information as bool field
3. Use the field when setting up the watchers in the corresponding `SetupWithManager` functions
4. Pass the information down into the route-checks `CheckBackendIsExistingService` & `CheckHasServiceImportSupport` (as these need this information). This might be a little bit fiddly - but should be possible (see below).

**Proposal for the checks**

Change their signature to 

```
func CheckHasServiceImportSupport(serviceImportEnabled bool) func(input Input) (bool, error) {
```

and change the call accordingly - by passing the field from the reconciler.

```
routechecks.CheckHasServiceImportSupport(r.serviceImportEnabled),
```



wdyt? It would be really great if you're up for this. Otherwise I can refactor this in a follow up.

Reply:
Yeah I wasn't sure about the global variable either...

I'm wondering if it's not better to like register the scheme only if the ServiceImport is installed and then have an helper function taking a client that would most likely do `client.Scheme().IsGroupRegistered()`. That way we don't have to fiddle some variable around and it's somewhat already handled by the client, WDYT?

Reply:
Great idea - haven't thought of that!

Even though we can't check the scheme for a Kind - checking for the group should be sufficient as we're in control when we add the group to the scheme (once we detected the `ServiceImport`).

And it would be still necessary to pass the scheme to the route-checks. But at least the scheme is already part of the reconcilers :+1: 

As long as we can get rid of the global var (and don't have to do remote calls) - go for it! Thanks!

Reply:
And done :tada: 

It was not too involved this way as I had access to a client everywhere I was supposed to check if ServiceImport is supported :+1: .

Reply:
Great! Yep wasn't aware that `Input` contains the `Client` :tada: